rust-runtime/aws-smithy-runtime/src/client/retries/strategy/standard.rs

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::client::interceptors::RequestAttempts;
+use aws_smithy_runtime_api::client::interceptors::InterceptorContext;
+use aws_smithy_runtime_api::client::orchestrator::{BoxError, ConfigBagAccessors};
+use aws_smithy_runtime_api::client::retries::{
+    ClassifyRetry, RetryReason, RetryStrategy, ShouldAttempt,
+};
+use aws_smithy_runtime_api::config_bag::ConfigBag;
+use std::time::Duration;
+
+const DEFAULT_MAX_ATTEMPTS: u32 = 4;
+
+#[derive(Debug)]
+pub struct StandardRetryStrategy {
+    max_attempts: u32,
+    initial_backoff: Duration,
+    max_backoff: Duration,
+    base: fn() -> f64,
+}
+
+impl StandardRetryStrategy {
+    pub fn with_base(mut self, base: fn() -> f64) -> Self {
+        self.base = base;
+        self
+    }
+
+    pub fn with_max_attempts(mut self, max_attempts: u32) -> Self {
+        self.max_attempts = max_attempts;
+        self
+    }
+
+    pub fn with_initial_backoff(mut self, initial_backoff: Duration) -> Self {
+        self.initial_backoff = initial_backoff;
+        self
+    }
+}
+
+impl Default for StandardRetryStrategy {
+    fn default() -> Self {
+        Self {
+            max_attempts: DEFAULT_MAX_ATTEMPTS,
+            max_backoff: Duration::from_secs(20),
+            // by default, use a random base for exponential backoff
+            base: fastrand::f64,
+            initial_backoff: Duration::from_secs(1),
+        }
+    }
+}
+
+impl RetryStrategy for StandardRetryStrategy {
+    // TODO(token-bucket) add support for optional cross-request token bucket
+    fn should_attempt_initial_request(&self, _cfg: &ConfigBag) -> Result<ShouldAttempt, BoxError> {
+        Ok(ShouldAttempt::Yes)
+    }
+
+    fn should_attempt_retry(
+        &self,
+        ctx: &InterceptorContext,
+        cfg: &ConfigBag,
+    ) -> Result<ShouldAttempt, BoxError> {
+        // Look a the result. If it's OK then we're done; No retry required. Otherwise, we need to inspect it
+        let output_or_error = ctx.output_or_error().expect(
+            "This must never be called without reaching the point where the result exists.",
+        );
```

Comment:
Are there retry strategies that don't require the output_or_error? Should we do the same wrapping strategy that we do for interceptors to make certain things always available in the context?